### ISSUE TITLE
Spack checksum improvement suggestions

### ISSUE THREAD

**Issue by HadrienG2** [spack/spack#8741]
- The hashing algorithm should be configurable (if it is, `spack checksum -h` says nothing about it)
- The default hashing algorithm should not be vulnerable to a major collision attack (unlike MD5 or SHA-1, the former of which I suspect to be the current default). SHA-256 is a popular choice these days.
- `spack checksum` should checksum the `resource()`s of a package as well.

**Comment by tgamblin** [comment]
Yes but the resource part is still a good idea. I think it probably deserves its own issue.
